runner/test/src/mill/runner/MillMainTests.scala

```diff
@@ -0,0 +1,95 @@
+package mill.runner
+
+import utest._
+
+object MillMainTests extends TestSuite {
+
+  private def assertParseErr(result: Either[String, Int], msg: String): Unit = {
+    assert(result.isLeft)
+    assert(result.swap.toOption.get.contains(msg))
+  }
+
+  def tests: Tests = Tests {
+
+    test("Parsing --jobs/-j flag") {
+
+      test("parse none") {
+        assert(MillMain.parseThreadCount(None, 10) == Right(10))
+      }
+
+      test("parse int number") {
+        assert(MillMain.parseThreadCount(Some("1"), 10) == Right(1))
+        assert(MillMain.parseThreadCount(Some("11"), 10) == Right(11))
+
+        assertParseErr(MillMain.parseThreadCount(Some("1.0"), 10), "Failed to find a int number")
+        assertParseErr(MillMain.parseThreadCount(Some("1.1"), 10), "Failed to find a int number")
+        assertParseErr(MillMain.parseThreadCount(Some("0.1"), 10), "Failed to find a int number")
+        assertParseErr(
+          MillMain.parseThreadCount(Some("0"), 10),
+          "Calculated cores to use should be a positive number."
+        )
+        assertParseErr(
+          MillMain.parseThreadCount(Some("-1"), 10),
+          "Calculated cores to use should be a positive number."
+        )
+      }
+
+      test("parse fraction number") {
+        assert(MillMain.parseThreadCount(Some("0.5C"), 10) == Right(5))
+        assert(MillMain.parseThreadCount(Some("0.54C"), 10) == Right(5))
+        assert(MillMain.parseThreadCount(Some("0.59C"), 10) == Right(5))
+        assert(MillMain.parseThreadCount(Some(".5C"), 10) == Right(5))
+        assert(MillMain.parseThreadCount(Some("1.0C"), 10) == Right(10))
+        assert(MillMain.parseThreadCount(Some("1.5C"), 10) == Right(15))
+
+        assertParseErr(
+          MillMain.parseThreadCount(Some("0.09C"), 10),
+          "Calculated cores to use should be a positive number"
+        )
```

Comment:
For these computed cases (this one and `C-10` and `C-11` below), let's always just round up to `1` core minimum. That's probably what users want, so no point erroring out when we can just do it for them